contracts/src/v0.1/RequestResponseConsumerBase.sol

```diff
@@ -22,16 +22,71 @@ abstract contract RequestResponseConsumerBase {
      * @return req request in memory
      */
     function buildRequest(bytes32 jobId) internal view returns (Orakl.Request memory req) {
-        return req.initialize(jobId, address(COORDINATOR), COORDINATOR.fulfillDataRequest.selector);
+        return
+            req.initialize(
+                jobId,
+                address(COORDINATOR),
+                COORDINATOR.fulfillDataRequestUint256.selector
+            );
     }
 
-    function fulfillDataRequest(uint256 requestId, uint256 response) internal virtual;
+    function fulfillDataRequestUint256(uint256 requestId, uint256 response) internal virtual;
 
-    function rawFulfillDataRequest(uint256 requestId, uint256 response) external {
+    function fulfillDataRequestInt256(uint256 requestId, int256 response) internal virtual;
+
+    function fulfillDataRequestBool(uint256 requestId, bool response) internal virtual;
+
+    function fulfillDataRequestString(uint256 requestId, string memory response) internal virtual;
+
+    function fulfillDataRequestBytes32(uint256 requestId, bytes32 response) internal virtual;
+
+    function fulfillDataRequestBytes(uint256 requestId, bytes memory response) internal virtual;
+
+    function rawFulfillDataRequestUint256(uint256 requestId, uint256 response) external {
```

Comment:
Wouldn't it be better to have a single modifier that takes care of the response verification so we do not have to reimplement the same stuff again and again? 🤔 

```Solidity
modifier verifyRawFulfillment {
    address coordinatorAddress = address(COORDINATOR);
    if (msg.sender != coordinatorAddress) {
        revert OnlyCoordinatorCanFulfill(msg.sender, coordinatorAddress);
    }
    _;
}
function rawFulfillDataRequestUint256(uint256 requestId, uint256 response) external verifyRawFulfillment {
    fulfillDataRequestUint256(requestId, response);
}
```